File: frontend/src/api/notice.js
import http from './http'

export const getNotices = (params) => http.get('/notices', { params })
export const getNotice = (id) => http.get(`/notices/${id}`)
export const createNotice = (data) => http.post('/notices', data)
export const updateNotice = (id, data) => http.put(`/notices/${id}`, data)
export const deleteNotice = (id) => http.delete(`/notices/${id}`)
export const submitNotice = (id, data) => http.put(`/notices/${id}/submit`, data)
export const withdrawNotice = (id) => http.put(`/notices/${id}/withdraw`, {})
export const approveNotice = (id, data) => http.put(`/notices/${id}/approve`, data)
export const cancelApproveNotice = (id) => http.put(`/notices/${id}/cancel-approve`, {})

export const uploadImage = (formData) => http.post('/upload/image', formData, {
  headers: { 'Content-Type': 'multipart/form-data' }
})
export const uploadAttachment = (formData) => http.post('/upload/attachment', formData, {
  headers: { 'Content-Type': 'multipart/form-data' }
})

File: backend/routes/notice.go
package routes

import (
	"oa-system/handlers"

	"github.com/gin-gonic/gin"
)

// noticeRoutes 执行相关业务逻辑
func noticeRoutes(rg *gin.RouterGroup) {
	rg.GET("/notices", handlers.GetNotices)
	rg.GET("/notices/:id", handlers.GetNotice)
	rg.POST("/notices", handlers.CreateNotice)
	rg.PUT("/notices/:id", handlers.UpdateNotice)
	rg.PUT("/notices/:id/submit", handlers.SubmitNotice)
	rg.PUT("/notices/:id/withdraw", handlers.WithdrawNotice)
	rg.PUT("/notices/:id/approve", handlers.ApproveNotice)
	rg.PUT("/notices/:id/cancel-approve", handlers.CancelApproveNotice)
	rg.DELETE("/notices/:id", handlers.DeleteNotice)
}
